src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt

```diff
@@ -0,0 +1,303 @@
+    summary = "Delete Ingress Event",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Event deleted"
+      )
+    ]
+  )
+  suspend fun deleteIngressEvent(
+    @Schema(description = "Event ID", required = true)
+    @PathVariable id: UUID,
+  ) = ingressEventDataRepository.deleteById(id)
+
+  @GetMapping("/ingressType")
+  @Operation(
+    summary = "Get Ingress Event Types",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Ingress Event Type"
+      )
+    ]
+  )
+  suspend fun getIngressEventTypes() = ingressEventTypeRepository.findAll().toList()
+
+  @DeleteMapping("/ingressType/{id}")
+  @Operation(
+    summary = "Delete Ingress Type",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Type deleted"
+      )
+    ]
+  )
+  suspend fun deleteIngressEventType(
+    @Schema(description = "Type ID", required = true)
+    @PathVariable id: String,
+  ) = ingressEventTypeRepository.deleteById(id)
+
+  @GetMapping("/publishers")
+  @Operation(
+    summary = "Get Publishers",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Publishers"
+      )
+    ]
+  )
+  suspend fun getPublishers() = publisherRepository.findAll().toList()
+
+  @DeleteMapping("/publishers/{id}")
+  @Operation(
+    summary = "Delete Publisher",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Publisher deleted"
+      )
+    ]
+  )
+  suspend fun deletePublisher(
+    @Schema(description = "Publisher ID", required = true)
+    @PathVariable id: UUID,
+  ) = publisherRepository.deleteById(id)
+
+  @GetMapping("/publisherSubscriptions")
+  @Operation(
+    summary = "Get Publisher Subscriptions",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Publisher Subscriptions"
+      )
+    ]
+  )
+  suspend fun getPublisherSubscriptions() = publisherSubscriptionRepository.findAll().toList()
+
+  @DeleteMapping("/publisherSubscriptions/{id}")
+  @Operation(
+    summary = "Delete Publisher Subscription",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Publisher Subscription deleted"
+      )
+    ]
+  )
+  suspend fun deletePublisherSubscription(
+    @Schema(description = "Publisher Subscription ID", required = true)
+    @PathVariable id: UUID,
+  ) = publisherSubscriptionRepository.deleteById(id)
+
+  //The below endpoints don't exist through any other means, so have added them here
+  @PostMapping("/ingressType")
```

Comment:
I think we need to make sure we've fully aligned around "ingress", "publisher" etc - but that's out of this scope I think

Reply:
I totally agree, this is just to keep the mapping consistent atm :)